Assets/LeapMotionModules/UI/Interaction/InteractionHand.cs

```diff
@@ -0,0 +1,780 @@
+                           Func<Hand> handAccessor,
+                           Chirality handedness,
+                           float hoverActivationRadius,
+                           float touchActivationRadius) {
+      this.interactionManager = interactionManager;
+      _handAccessor = handAccessor;
+      _isLeft = handedness == Chirality.Left;
+
+      InitHovering(hoverActivationRadius);
+      InitTouch(touchActivationRadius);
+      InitContact();
+      InitGrasping();
+    }
+
+    public Hand GetLeapHand() {
+      return _hand;
+    }
+
+    public void FixedUpdateHand(bool doHovering, bool doContact, bool doGrasping) {
+      _hand = _handAccessor();
+
+      if (doHovering) FixedUpdateHovering();
+      if (doContact || doGrasping) FixedUpdateTouch();
+      if (doContact) FixedUpdateContact();
+      if (doGrasping) FixedUpdateGrasping();
+    }
+
+    #region Hovering
+
+    private void InitHovering(float hoverActivationRadius) {
+      // Standard Hover
+      _hoverActivityManager = new ActivityManager(interactionManager, hoverActivationRadius);
+    }
+
+    private void FixedUpdateHovering() {
+      _hoverActivityManager.FixedUpdateHand(_hand);
+      CalculateIntentionPosition();
+      HoverCheckResults hoverResults = CheckHoverForHand(_hand, _hoverActivityManager.ActiveBehaviours);
+      ProcessHoverCheckResults(hoverResults);
+      ProcessPrimaryHoverCheckResults(hoverResults);
+    }
+
+    /// <summary>
+    /// Encapsulates tracking Hand <-> InteractionBehaviour hover candidates
+    /// (broad-phase) via PhysX sphere queries.
+    /// </summary>
+    private ActivityManager _hoverActivityManager;
+    public float HoverActivationRadius {
+      get {
+        return _hoverActivityManager.activationRadius;
+      }
+      set {
+        _hoverActivityManager.activationRadius = value;
+      }
+    }
+
+    private struct HoverCheckResults {
+      public HashSet<InteractionBehaviourBase> hovered;
+      public InteractionBehaviourBase primaryHovered;
+      public float primaryHoveredDistance;
+      public Hand checkedHand;
+    }
+
+    private HashSet<InteractionBehaviourBase> _hoveredLastFrame = new HashSet<InteractionBehaviourBase>();
+
+    private HashSet<InteractionBehaviourBase> _hoverableCache = new HashSet<InteractionBehaviourBase>();
+    private HoverCheckResults CheckHoverForHand(Hand hand, HashSet<InteractionBehaviourBase> hoverCandidates) {
+      _hoverableCache.Clear();
+
+      HoverCheckResults results = new HoverCheckResults() {
+        hovered = _hoverableCache,
+        primaryHovered = null,
+        primaryHoveredDistance = float.PositiveInfinity,
+        checkedHand = hand
+      };
+
+      foreach (var interactionObj in hoverCandidates) {
+        results = CheckHoverForElement(hand, interactionObj, results);
+      }
+
+      return results;
+    }
+
+    private HoverCheckResults CheckHoverForElement(Hand hand, InteractionBehaviourBase hoverable, HoverCheckResults curResults) {
```

Comment:
Looks like this method is returning the argument, but modified.  A better way would be to pass in curResults as a ref param, which makes it more efficient, but also more obvious what it is doing.